tests/test_cli.py

```diff
@@ -37,8 +38,109 @@ def test_main(capsys, expected_dir, args):
         "todo.pyi",
         "todo_v2.pyi",
     ]
+
+    # Check that files match expected output
     match, mismatch, errors = filecmp.cmpfiles(output_dir, expected_dir, pyi_files)
     assert errors == []
     assert mismatch == []
     assert match == pyi_files
     shutil.rmtree(output_dir)
+
+
+def test_generated_code_is_importable(tmp_path):
+    """Test that generated code can actually be imported and used."""
+    input_dir = "example/interfaces"
+    output_dir = tmp_path / "generated"
+
+    # Generate stubs with frozen option (without kw_only)
+    main([input_dir, "--output", str(output_dir), "--frozen"])
+
+    # Copy .pyi files to .py files so they can be imported
+    for pyi_file in output_dir.glob("*.pyi"):
+        py_file = pyi_file.with_suffix(".py")
+        py_file.write_text(pyi_file.read_text())
+
+    # Create __init__.py to make it a proper package
+    (output_dir / "__init__.py").write_text("")
+
+    # Create a test script that imports and uses the generated code
+    test_script = tmp_path / "test_import.py"
+    test_script.write_text(
+        """
+import sys
+from pathlib import Path
+sys.path.insert(0, str(Path(__file__).parent))
+
+# Import the generated package (not individual modules)
+from generated import todo
+
+# Try to instantiate TodoItem with defaults
+# This will raise NameError if the lambda references undefined DateTime
+item = todo.TodoItem()
+print("Success: TodoItem instantiated")
+"""
+    )
+
+    # Run the test script
+    import subprocess
+
+    result = subprocess.run(
+        [sys.executable, str(test_script)], capture_output=True, text=True
+    )
+
+    # Check for NameError specifically
+    assert (
+        "NameError" not in result.stderr
+    ), f"Generated code has undefined names: {result.stderr}"
+    assert result.returncode == 0, f"Failed to run generated code: {result.stderr}"
+    assert "Success" in result.stdout
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="kw_only requires Python 3.10+")  # pragma: no cover
+def test_generated_code_with_kw_only_is_importable(tmp_path):  # pragma: no cover
```

Comment:
The code coverage tool runs on every single Python version. On everything 3.10+ it is 100%, but on 3.9 it is 70%, which fails. Just skipping this test as far as code coverage, since the other test runs for every Python version.

Net net - this test DOES run when expected, and failures ARE caught.